popup2.js

```diff
@@ -1,8 +1,48 @@
-window.addEventListener('load',function(){
-	document.getElementById('submitbutton').addEventListener("click", function() {
+window.addEventListener('load',function() {
+	document.getElementById('submitbutton').addEventListener('click', function() {
 		var firstName = document.getElementById('firstName')
 		var middleName = document.getElementById('middleName')
 		var lastName = document.getElementById('lastName')
 		var domain = document.getElementById('domain') 
-	} );
-} );
+		var permutations = getPermutations(firstName, middleName, lastName, domain);
+	});
+});
+
+function getPermutations(firstName, middleName, lastName, domain) {
+	/* Returns all of the desired permutations for the given data */
+
+	// Extract the needed variables
+	var firstInitial = firstName[0]
+	
+	// This list is incomplete, but should be sufficient to get started
+	var permutationTemplates = [
+		'{fn}',
+		'{ln}',
+		'{fn}{ln}',
+		'{fn}.{ln}',
+		'{fi}{ln}'
+	]
```

Comment:
We can use the complete list later, but this will be useful for now and you can add more as you add different variables (like lastInitial, middleInitial,...)